src/colmap/estimators/cost_functions.h

```diff
@@ -88,34 +144,42 @@ class ReprojErrorCostFunction {
                             &residuals[1]);
     residuals[0] -= T(observed_x_);
     residuals[1] -= T(observed_y_);
+    if constexpr (CTYPE != CovarianceType::IDENTITY) {
```

Comment:
This if condition is redundant to the implementation inside ApplyCovariance?

Reply:
I felt that adding this if makes the code more readable. And since if constexpr is handled at compilation, it saves a slight overhead of calling the function.

Reply:
Since the function call is inlined (even though not 100% guaranteed), it should result in the same generated code?

Reply:
Sure. This is done.